File: cucumber_net/chapter02/FirstTaste/FirstTaste/Steps/Chapter02/加法Steps.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using TechTalk.SpecFlow;

namespace FirstTaste.Steps.Chapter02
{
    [Binding]
    public class 加法Steps {
        private string _input;
        private int _output;

        [Given(@"输入 ""(.*)""")]
        public void 假如输入(string input) {
            _input = input;
        }
        
        [When(@"运行计算器")]
        public void 当运行计算器()
        {
            _output = new Calc().Add(_input);
        }
        
        [Then(@"输出 ""(.*)""")]
        public void 那么输出(int expectedOutput)
        {
            Assert.AreEqual(expectedOutput, _output);
        }
    }
}

File: cucumber_javascript/chapter02/features/stepdefinitions/加法.steps.js
'use strict'

var assert = require('assert');

module.exports = function () {
    var _input;
    var _output;
    this.Given(/^输入 "([^"]*)"$/, function (input, callback) {
        _input = input;
        callback();
    });

    this.When(/^运行计算器$/, function (callback) {
        _output = eval(_input);
        callback();
    });

    this.Then(/^输出 "([^"]*)"$/, function ( output, callback) {
        assert.equal(output, _output);
        callback();
    });
};